### ISSUE TITLE
zig build autograd-test fails at harness level (--listen=- test-runner SIGABRT); tests pass directly

### ISSUE THREAD

**Issue by miruamel** [miruamel/baiyi#27]
## Symptom
`zig build autograd-test` reports a `failed command` and the gate is red, but the autograd tests are actually correct.

## Evidence
- Running the built test binary **directly** returns `RC=0` and prints `All 9 tests passed.` (xor MLP, 3-class classifier, gradient-formula checks, matmul/softmax/cross-entropy finite differences, Adam steps).
- Running the same binary with the exact flags `zig build` passes — `./<bin> --cache-dir=./.zig-cache --seed=0x... --listen=-` — returns `RC=134` (SIGABRT) with:
  ```
  return mainServer(init) catch |err| panic("internal test runner failure: {t}", .{err});
  ```
- Verified **pre-existing on clean `main`** (stashed all local changes, reproduced identically), so it is not introduced by any recent feature work and is orthogonal to `tile` (#26), `unsqueeze`/`squeeze` (#21), etc.

## Analysis
The autograd *code* is green; the failure is in how `zig build` launches the test artifact. Both the `test` and `autograd-test` steps use `b.addRunArtifact(...)` in `build.zig`, yet `zig build test` passes while `zig build autograd-test` panics at startup under `--listen=-`. A naive fix (`autograd_tests.run.step`) fails to compile — `Build.Step.Compile` has no `run` field in Zig 0.16.

## Impact
The `autograd-test` gate cannot be used to verify autograd changes via `zig build`. Correctness is still verifiable by running the built test binary directly (it passes 9/9).

## Suggested direction
Investigate the Zig 0.16 build-system test-runner invocation: why `zig build` passes an invalid `--listen=-` to the autograd test binary but a working address to the main test binary, or wire the autograd step via the canonical test-run mechanism for this Zig version.
